lib/rubocop/cop/rails/time_zone.rb

```diff
@@ -106,10 +106,17 @@ def extract_method(node)
         end
 
         # Only add the method to the chain if the method being
-        # called if it's a member of the time class.
-        def receiver_contains_time?(node)
+        # called is part of the time class.
+        def root_receiver_contains_time?(node)
           receiver, _method_name, *_args = *node
-          receiver.source.include? 'Time' 
+          time_classes = ['Time', 'DateTime', '::Time', '::DateTime']
+          contains_time = time_classes.include? receiver.source
+          receiver.children.each do |child|
+            if child.respond_to? 'source'
```

Comment:
I still don't get why do we need to be doing source checks? Can't you just find the top receiver and check what it is?